src/Parentheses.cpp

```diff
@@ -1,5 +1,41 @@
 #include "Parentheses.hpp"
+#include <stack>
+#include <map>
+#include <algorithm>
+#include <iostream>
 
-bool Parentheses::isBalanced(std::string s) {
-    return true;
+bool Parentheses::isBalanced(std::string stringInput) {
+    std::map<char, char> bracketsMap =
+    {
+        {'(', ')'},
+        {'[', ']'},
+        {'{', '}'},
+    };
+    
+    std::stack<char> brackets;
+    for(auto strItr=stringInput.begin(); strItr!=stringInput.end(); ++strItr)
+    {
+        auto strItrd = *strItr;
```

Comment:
nazwy brzydkie. Jednak podobna do drugiej i niewiele mówiące. Łatwo się pomylić. Lepiej je zmienić. A w ogóle to chyba pętla zakresowa for po kolekcji będzie lepsza.